src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,538 @@
+
+```ts title="src/lib/contentful.ts" ins={2,4-10}
+import contentful from "contentful";
+import type { Document } from "@contentful/rich-text-types";
+
+export interface BlogPost {
+  title: string;
+  date: string;
+  description: string;
+  content: Document;
+  slug: string;
+}
+
+export const contentfulClient = contentful.createClient({
+  space: import.meta.env.CONTENTFUL_SPACE_ID,
+  accessToken: import.meta.env.DEV
+    ? import.meta.env.CONTENTFUL_PREVIEW_TOKEN
+    : import.meta.env.CONTENTFUL_DELIVERY_TOKEN,
+  host: import.meta.env.DEV ? "preview.contentful.com" : "cdn.contentful.com",
+});
+
+```
+
+Next, go to your `index.astro` file in `src/pages/` and import your `BlogPost` interface and  `contentfulClient` from `src/lib/contentful.ts`. 
+
+Using the `getEntries` method from `contentfulClient`, fetch all the entries with a content type of `blogPost`. Pass the `BlogPost` interface to the `getEntries` generic to type your response. 
+
+```astro title="src/pages/index.astro"
+---
+import { contentfulClient } from "../lib/contentful";
+import type { BlogPost } from "../lib/contentful";
+
+const entries = await contentfulClient.getEntries<BlogPost>({
+  content_type: "blogPost",
+});
+---
+```
+
+The items property of `entries` is an array of blog posts. You can use `entries.items.map` to iterate over the array and format the response. In this example, we will destructure the `fields` property of each item and reformat the date to a more readable format.
+
+```astro title="src/pages/index.astro" ins={9-17}
+---
+import { contentfulClient } from "../lib/contentful";
+import type { BlogPost } from "../lib/contentful";
+
+const entries = await contentfulClient.getEntries<BlogPost>({
+  content_type: "blogPost",
+});
+
+const posts = entries.items.map((item) => {
+  const { title, date, description, slug } = item.fields;
+  return {
+    title,
+    slug,
+    description,
+    date: new Date(date).toLocaleDateString()
+  };
+});
+---
+```
+
+Finally, you can use `posts` in your template to show a preview of each blog post.
+
+```astro astro title="src/pages/index.astro" ins={19-37}
+---
+import { contentfulClient } from "../lib/contentful";
+import type { blogPost } from "../lib/contentful";
+
+const entries = await contentfulClient.getEntries<blogPost>({
+  content_type: "blogPost",
+});
+
+const posts = entries.items.map((item) => {
+  const { title, date, description, slug } = item.fields;
+  return {
+    title,
+    slug,
+    description,
+    date: new Date(date).toLocaleDateString()
+  };
+});
+---
+<html lang="en">
+  <head>
+    <title>My Blog</title>
+  </head>
+  <body>
+    <h1>My Blog</h1>
+    <ul>
+      {posts.map((post) => (
+        <li>
+          <a href={`/posts/${slug}/`}>
+            <h2>{title}</h2>
+          </a>
+          <time>{date}</time>
+          <p>{description}</p>
+        </li>
+      ))}
+    </ul>
+  </body>
+</html>
+```
+
+### Dynamic routes
```

Comment:
```suggestion
### Dynamic routes

Use the same method to fetch data your data from Contentful as above, but this time, on a page that will create a unique page route for each blog post.

```
I think it might be useful to point out somehow that a lot of this will look familar, since the fetch call is very similar! This also allows us to do some referring back vs reiterate the exact same thing again.